worker/src/cache.ts

```diff
@@ -110,3 +115,152 @@ export function getCacheSettings(
     };
   }
 }
+
+async function serializeResponse(response: Response): Promise<string> {
+  const serializableResponse = {
+    body: await response.text(), // or .arrayBuffer() if you expect binary content
+    status: response.status,
+    headers: Object.fromEntries(response.headers.entries()),
+  };
+
+  return JSON.stringify(serializableResponse);
+}
+
+function deserializeResponse(serializedResponse: string): Response {
+  const responseObj = JSON.parse(serializedResponse);
+
+  return new Response(responseObj.body, {
+    status: responseObj.status,
+    headers: responseObj.headers,
+  });
+}
+
+async function buildCachedRequest(
+  request: Request,
+  idx: number
+): Promise<string> {
+  const headers = new Headers();
+  for (const [key, value] of request.headers.entries()) {
+    if (key.toLowerCase().startsWith("helicone-")) {
+      headers.set(key, value);
+    }
+    if (key.toLowerCase() === "authorization") {
+      headers.set(key, value);
+    }
+  }
```

Comment:
Can we change this to just authorization and helicone-auth headers
```suggestion
  for (const [key, value] of request.headers.entries()) {
    if (key.toLowerCase().startsWith("helicone-auth")) {
      headers.set(key, value);
    }
    if (key.toLowerCase() === "authorization") {
      headers.set(key, value);
    }
  }
```

Reply:
done!